layers/ip6.go

```diff
@@ -280,165 +229,191 @@ func decodeIPv6(data []byte, p gopacket.PacketBuilder) error {
 	return p.NextDecoder(ip6.NextHeader)
 }
 
-func (i *IPv6HopByHop) SerializeTo(b gopacket.SerializeBuffer, opts gopacket.SerializeOptions) error {
-	var bytes []byte
-	var err error
+type IPv6HeaderTLVOption interface {
```

Comment:
The IPv6HeaderTLVOption interface doesn't have to be exported but it does allow users of the API to implement their own TLV types. Right? Though it is possible to use just the IPv6HeaderTLVOptionUnknown type.

It doesn't have a DecodeFromBytes() method, though I suppose it could, with a suitable type code table. Unless I'm missing something.

Dunno about the use of malloc. I feel like the code is more idiomatic Go and it shows. It doesn't treat IPv6HopByHopOption and IPv6DestinationOption as different objects. There are no BigEndian.Uint32() calls to set/get the values (for Pad and Jumbo at least). And the code to align and serialize the TLVs is also simpler and perhaps more elegant. It computes the length of the aligned/padded HopByHop/Destination header without a serialize dry run. Is that issue a showstopper?